### PR TITLE
Fix(email) prevent UDF forms from sending attachments to submitted email addresses

### PR DESCRIPTION
**Pull request by stephenmcm**

### CODE REVIEW

File: _config/config.yml [hunk 1]
```diff
@@ -41,6 +41,24 @@ FileField:
...
+UserForm:
+  extensions:
+    - BasisUserFormSpamProtectionExtension
+Form:
+  extensions:
+    - BasisFormProtectionExtension
+BasisFormProtectionExtension:
+  controllers:
+    - Page_Controller
+---
```
Comment (stephenmcm) [inline _config/config.yml]: All the above should be excluded shouldn't it?

Comment (nyeholt) [reply _config/config.yml]: Yeah, where is this coming into it from?

Comment (nglasl) [reply _config/config.yml]: Isn't that stuff from the PR that @Neumes has pending?

File: code/extensions/UserDefinedForm_ControllerAttachmentFilter.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * Fixes security risk of emailing attachments to publicly submitted email addresses
+ *
+ * @author stephen
+ */
+class UserDefinedForm_ControllerAttachmentFilter extends DataExtension {
```
Comment (nglasl) [inline code/extensions/UserDefinedForm_ControllerAttachmentFilter.php]: Just one more thing, sorry @stephenmcm. Can you please add `Extension` into the class name (and file name) as previously discussed?

**Comment by nyeholt** [comment]
Also, it should extend Extension, not dataextension, which should only be
used to extend a dataobject

**Pull request merged**